lib/college/schemas/student.ex

```diff
@@ -16,9 +28,20 @@ defmodule College.Schemas.Student do
     timestamps()
   end
 
+  def fill_virtual_fields(%Student{} = student) do
+    student = student |> Repo.preload(:courses)
+
+    num =
+      student.courses
+      |> Enum.filter(fn course -> course.semester == :first end)
+      |> length()
+
+    Map.put(student, :number_of_courses, num)
+  end
+
```

Comment:
This logic should be private and moved inside the protocol, since it's related to Person implementation 
student
|> Repo.preload(:courses)
|> ...